avocado/loader.py

```diff
@@ -53,19 +54,52 @@ def _make_simple_test(self, test_path, params):
     def _make_test(self, test_name, test_path, params, queue):
         module_name = os.path.basename(test_path).split('.')[0]
         test_module_dir = os.path.dirname(test_path)
+        test_class = None
         try:
             f, p, d = imp.find_module(module_name, [test_module_dir])
             test_module = imp.load_module(module_name, f, p, d)
             f.close()
-            test_class = getattr(test_module, module_name)
-        except ImportError:
-            test_class = test.MissingTest
-        finally:
-            test_parameters = {'name': test_name,
-                               'base_logdir': self.job.logdir,
-                               'params': params,
-                               'job': self.job,
-                               'runner_queue': queue}
+            for name, obj in inspect.getmembers(test_module):
+                if inspect.isclass(obj):
+                    if issubclass(obj, test.Test):
+                        test_class = obj
+            if test_class is not None:
+                test_parameters = {'name': test_name,
+                                   'base_logdir': self.job.logdir,
+                                   'params': params,
+                                   'job': self.job,
+                                   'runner_queue': queue}
+            else:
+                if os.access(test_path, os.X_OK):
+                    test_class = test.SimpleTest
+                    test_parameters = {'path': test_path,
+                                       'base_logdir': self.job.logdir,
+                                       'params': params,
+                                       'job': self.job}
+                else:
+                    test_class = test.NotATest
+                    test_parameters = {'name': test_name,
+                                       'base_logdir': self.job.logdir,
+                                       'params': params,
+                                       'job': self.job,
+                                       'runner_queue': queue}
+
+        except ImportError, details:
+            if os.access(test_path, os.X_OK):
```

Comment:
Yes, there is the case where it is a Python source and executable, like pdb.py and tabnanny.py on python modules. It's correct to assume this.